src/lib/client/Client.ts

```diff
@@ -233,6 +241,28 @@ export class Client extends BaseClient {
 		return this;
 	}
 
+	public sweepMessages(lifetime = this.options.cache.messageLifetime): number {
+		if (typeof lifetime !== 'number' || isNaN(lifetime)) throw new TypeError('The lifetime must be a number.');
+		if (lifetime <= 0) {
+			this.emit(ClientEvents.Debug, 'Didn\'t sweep messages - lifetime is unlimited');
+			return -1;
+		}
+
+		const now = Date.now();
+		let channels = 0;
+		let messages = 0;
+
+		for (const channel of this.channels.values()) {
+			if (!isTextBasedChannel(channel)) continue;
+			channels++;
+
+			messages += channel.messages.sweep(message => now - (message.editedTimestamp || message.createdTimestamp) > lifetime);
+		}
```

Comment:
This would be more a far more efficient alternative:
```ts
for (const guild of this.guilds.values()) {
	for (const channel of guild.channels.values()) {
		++channels;
		messages += channel.messages.sweep(message => now - (message.editedTimestamp || message.createdTimestamp) > lifetime);
	}
}
```

Reply:
why? then you miss dm channels...

Reply:
Same behaviour as the current code (see line 256).

```ts
if (!isTextBasedChannel(channel)) continue;
```

Reply:
well that behavior is wrong then... It should sweep all channels where there is a messages cache.

Reply:
correction, the code is right, https://github.com/dirigeants/core/blob/master/src/lib/util/Util.ts#L26